tensorflow/lite/micro/kernels/if.cc

```diff
@@ -86,11 +89,10 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
   const OpData* op_data = reinterpret_cast<OpData*>(node->user_data);
 
   tflite::MicroContext* micro_context = tflite::GetMicroContext(context);
-  TfLiteTensor* cond = micro_context->AllocateTempInputTensor(node, 0);
-
+  const TfLiteEvalTensor* cond =
+      tflite::micro::GetEvalInput(context, node, kCondTensor);
   TF_LITE_ENSURE(context, cond != nullptr);
-  bool cond_value = cond->data.b[0];
-  micro_context->DeallocateTempTfLiteTensor(cond);
```

Comment:
Don't we still need to deallocate the `cond` Tensor that was allocated?

Reply:
TfLiteEvalTensor is a permanent allocation that exists for the lifetime of the MicroInterpreter.

Reply:
Ah sorry, I misread. I got it confused with the `cond` allocated at line 58.